### PR TITLE
fix(js): format .lib.swcrc file with nx format

### LINKED ISSUE #10240: nx format:write doesn't format .swcrc files
## Current Behavior

Running `nx format:write` does not format `.swcrc` files (e.g. `libs/my-lib/.lib.swcrc`).

## Expected Behavior

Any format issues in a `.swcrc` file (e.g. incorrect tab spacing) should be addressed when running `nx format:write`.

## Steps to Reproduce

1. Introduce a format issue into a `.swcrc` file (e.g. change the tab spacing).
2. Run `nx format:write`.
3. Notice that the lint issue hasn't been fixed.

### Failure Logs

N/A

### Environment

```
Node : 16.14.2
OS   : darwin x64
npm  : 8.5.0

nx : 14.1.4
@nrwl/angular : Not Found
@nrwl/cypress : 14.1.4
@nrwl/detox : Not Found
@nrwl/devkit : 14.1.4
@nrwl/eslint-plugin-nx : 14.1.4
@nrwl/express : Not Found
@nrwl/jest : 14.1.4
@nrwl/js : 14.1.4
@nrwl/linter : 14.1.4
@nrwl/nest : Not Found
@nrwl/next : Not Found
@nrwl/node : Not Found
@nrwl/nx-cloud : 14.0.3
@nrwl/nx-plugin : Not Found
@nrwl/react : 14.1.4
@nrwl/react-native : Not Found
@nrwl/schematics : Not Found
@nrwl/storybook : 14.1.4
@nrwl/web : 14.1.4
@nrwl/workspace : 14.1.4
typescript : 4.5.5
rxjs : 6.6.7
---------------------------------------
Community plugins:
```

### PR DESCRIPTION
**Pull request by nartc**
ISSUES CLOSED: #10240

## Current Behavior

Running `nx format` does not format `.lib.swcrc`

## Expected Behavior

Running `nx format` should format `.lib.swcrc` files

This is tricky (and it's my fault for using `.lib.swcrc`). Prettier recognizes `.swcrc` as a File instead of as an Extension. In addition, adding `**/.*.swcrc` alone isn't enough.

## Related Issue(s)

Fixes #10240

### CODE REVIEW

File: packages/nx/src/command-line/format.ts [hunk 1]
```diff
@@ -83,10 +83,16 @@ async function getPatterns(
...
+
     const supportedExtensions = prettier
       .getSupportInfo()
       .languages.flatMap((language) => language.extensions)
-      .filter((extension) => !!extension);
+      .filter((extension) => !!extension)
+      // Prettier supports ".swcrc" as a file instead of an extension
+      // So we add ".swcrc" as a supported extension manually
+      // which allows it to be considered for calculating "patterns"
+      .concat('.swcrc');
```
Comment (AgentEnder) [inline packages/nx/src/command-line/format.ts:94]: I know we setup the json parser for the files when we generate the swcrc config, but if someone **_didnt_** use  a generator and had an swcrc file, what would happen? Is there any way for us to set the json parser here?

Comment (nartc) [reply packages/nx/src/command-line/format.ts:94]: That's a good point. Should we?

Comment (AgentEnder) [reply packages/nx/src/command-line/format.ts:94]: If we can set it at runtime here, I'd think that'd be nicer. Looking at this code vs. the `formatFiles` utility function, we aren't programmatically calling prettier. We are just using the CLI + child_process. That may make it hard or not feasible in this case.

That said, we probably want these changes in the `formatFiles` helper too, s.t. the files are formatted if a generator changes them.

Comment (nartc) [reply packages/nx/src/command-line/format.ts:94]: Tried it with runtime now. A little ugly but it's better than modifying the prettier config on generation (and modifying approach only works for .prettierrc in JSON format)

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**